engineering/conventions.md

```diff
@@ -63,6 +64,35 @@ Some projects might define both build and runtime requirements that go beyond wh
 
 * If it is idiomatic in your language, prefer early handling of errors and return early (more info: [Avoid Else, Return Early](http://blog.timoxley.com/post/47041269194/avoid-else-return-early), [GuardClause](http://wiki.c2.com/?GuardClause) and [HandleErrorsInContext](http://wiki.c2.com/?HandleErrorsInContext)).
 
+## TODO comments
+
+The TODO comments should match the following format:
+
+```
+// KEYWORD(reference): comment text
+```
+
+A single line comment token must be used (`//` in case of C/C++, Go, etc). This is done to allow commenting large chunks of code with multi-line comment tokens (`/*`) during development and refactoring.
+
+The `KEYWORD` must be one of the following:
+
+- `TODO` - Something is missing or is not up to standard, but at the same time is not critical for the current implementation. It may be a missing optimization, a possible new feature, a refactoring opportunity or a deprecation note.
```

Comment:
It may even be fine as written, but might require reconsideration (e.g., "Figure out whether we need to distinguish between subcategories of frozzle within each bucket"). I suggest not focusing on "deficiency" as the gauge (that's arguably more "FIXME") but the "future consideration". Perhaps:

> `TODO`: Highlights future work or unanswered questions that are outside the scope of the current PR, but that should be considered when modifying this code in the future. For example, a missing optimization, possible new features, something to refactor, or a deprecated usage to replace.

Or something along those lines?